src/processor/map/map_simple.cpp

```diff
@@ -67,16 +78,22 @@ std::unique_ptr<PhysicalOperator> PlanMapper::mapExportDatabase(
     if (fs->fileOrPathExists(filePath, clientContext)) {
         throw RuntimeException(stringFormat("Directory {} already exists.", filePath));
     }
+    fs->createDir(filePath);
     auto printInfo = std::make_unique<ExportDBPrintInfo>(filePath, boundFileInfo->options);
     auto messageTable =
         FactorizedTableUtils::getSingleStringColumnFTable(clientContext->getMemoryManager());
     auto exportDB = std::make_unique<ExportDB>(boundFileInfo->copy(),
         exportDatabase->isSchemaOnly(), messageTable, getOperatorID(), std::move(printInfo));
     auto sink = std::make_unique<DummySimpleSink>(messageTable, getOperatorID());
+    sink->addChild(std::move(exportDB));
     for (auto child : exportDatabase->getChildren()) {
         sink->addChild(mapOperator(child.get()));
     }
-    sink->addChild(std::move(exportDB));
+    if (boundFileInfo->fileTypeInfo.fileType == common::FileType::CSV) {
```

Comment:
You don't have to check whether this is csv or not, you can always init the parallelFlag in the exportdb sharedState..